homeassistant/components/airgradient/sensor.py

```diff
@@ -0,0 +1,139 @@
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda status: status.pm02,
+    ),
+    AirGradientSensorEntityDescription(
+        key="pm10",
+        device_class=SensorDeviceClass.PM10,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda status: status.pm10,
+    ),
+    AirGradientSensorEntityDescription(
+        key="temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda status: status.ambient_temperature,
+    ),
+    AirGradientSensorEntityDescription(
+        key="humidity",
+        device_class=SensorDeviceClass.HUMIDITY,
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda status: status.relative_humidity,
+    ),
+    AirGradientSensorEntityDescription(
+        key="signal_strength",
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda status: status.signal_strength,
+    ),
+    AirGradientSensorEntityDescription(
+        key="tvoc",
+        translation_key="total_volatile_organic_component_index",
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda status: status.total_volatile_organic_component_index,
+    ),
+    AirGradientSensorEntityDescription(
+        key="nitrogen_index",
+        translation_key="nitrogen_index",
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda status: status.nitrogen_index,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up AirGradient sensor entities based on a config entry."""
+
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        AirGradientSensor(coordinator, description) for description in SENSOR_TYPES
+    )
+
+
+class AirGradientSensor(
+    CoordinatorEntity[AirGradientDataUpdateCoordinator], SensorEntity
+):
+    """Defines an AirGradient sensor."""
+
+    _attr_has_entity_name = True
+
+    entity_description: AirGradientSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: AirGradientDataUpdateCoordinator,
+        description: AirGradientSensorEntityDescription,
+    ) -> None:
+        """Initialize airgradient sensor."""
+        super().__init__(coordinator)
+
+        self.entity_description = description
+        self._attr_unique_id = f"{coordinator.data.serial_number}-{description.key}"
+        self._attr_device_info = DeviceInfo(
```

Comment:
You are not setting model. It's available as per the zeroconf info.

Reply:
Also, for device name, in the past if we didn't get a name from the device, we've been using the config entry title. That way there is only a single place where one needs to set this data.